src/Core/CustomMethods.php

```diff
@@ -282,10 +282,15 @@ protected function removeMethodsFrom($property, $index = null)
                 if (!isset(self::$extra_methods[$class][$method])) {
                     continue;
                 }
-                    
+
                 $methodInfo = self::$extra_methods[$class][$method];
 
-                if ($methodInfo['property'] === $property && $methodInfo['index'] === $index) {
+                if (
+                    // always check for property
+                    (isset($methodInfo['property']) && $methodInfo['property'] === $property) &&
+                    // check for index only if provided (otherwise assume true)
+                    (($index && isset($methodInfo['index']) && $methodInfo['index'] === $index) || true)
```

Comment:
Is the `|| true` bit here for debugging?

Reply:
No, it still needs to unset the $class $method when index is not provided, but when it is, it needs to match the methodInfo part.

Reply:
That whole line is completely pointless then. You might as well remove it.

Reply:
You've essentially got; `stuff && (true || whatever-does-not-matter)`

Reply:
Yeap, I must have been brain tired end of last week, apologies.

Updated the logic to reflect the original intention that it either doesn't have the $index set or it checks it when it is set.